pkg/util/cache/intern.go

```diff
@@ -0,0 +1,359 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package cache
+
+import (
+	"fmt"
+	cconfig "github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	lru "github.com/hashicorp/golang-lru"
+	"go.uber.org/atomic"
+	"math"
+	"os"
+	"sync"
+	"time"
+	"unsafe"
+)
+
+// initialInternerSize is the size of the LRU cache (in #strings).  This is HEAP, so
+// don't let this get too big compared to the MMAP region.
+const initialInternerSize = 64
+
+// backingBytesPerInCoreEntry is the number of bytes to allocate in the mmap file per
+// element in our LRU.  E.g., some value of initialInternerSize * POW(growthFactor, N).
+const backingBytesPerInCoreEntry = 4096
+const growthFactor = 1.5
```

Comment:
I'd like to see this and `initialInternerSize` configurable.

Reply:
Perfectly fair.  Do you mind if we do a little bounds checking on the growth factor config?  <= 1 is catastrophically bad, >2 is probably also bad.  I chose 1.5 as it's the default in other standard libraries I've used in the past (e.g. it's the growth rate of `std::vector` in c++)

Reply:
Bounds checking makes sense to me, absolutely.